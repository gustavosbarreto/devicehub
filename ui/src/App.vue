<template>
<v-app>
    <v-navigation-drawer v-if="isLoggedIn" :clipped="false" fixed v-model="drawer" :mini-variant="false" enable-resize-watcher app>
        <v-container>
            <div class="text-center">
                <v-icon>mdi-console</v-icon>
                <h2 style="font-family: monospace">ShellHub</h2>
                <span class="overline">beta</span>
            </div>
        </v-container>
        <v-list>
            <v-list-item v-for="item in items" :key="item.title" :to="item.path" two-line>
                <v-list-item-action>
                    <v-icon v-text="item.icon"></v-icon>
                </v-list-item-action>

                <v-list-item-content>
                    <v-list-item-title v-text="item.title"></v-list-item-title>
                </v-list-item-content>
            </v-list-item>
        </v-list>
    </v-navigation-drawer>
    <v-app-bar app elevate-on-scroll color="primary" class="pl-3 pr-4" v-if="isLoggedIn">
        <v-menu transition="scale-transition" origin="top left">
            <template v-slot:activator="{ on }">
                <v-chip v-on="on">
                    <v-icon left>mdi-server</v-icon>
                    My Device Fleet
                    <v-icon right>mdi-chevron-down</v-icon>
                </v-chip>
            </template>
        </v-menu>
        <v-spacer></v-spacer>
        <v-chip>
            <v-icon>help</v-icon>
        </v-chip>
        <v-chip>
            <v-icon>notifications</v-icon>
        </v-chip>
        <v-menu transition="scale-transition" origin="top right">
            <template v-slot:activator="{ on }">
                <v-chip v-on="on">
                    <v-icon left>mdi-account</v-icon>
                    {{ $store.getters["auth/currentUser"] }}
                    <v-icon right>mdi-chevron-down</v-icon>
                </v-chip>
            </template>

            <v-card>
                <v-list-item three-line>
                    <v-list-item-content>
                        <v-list-item-title class="mb-1">Tenant ID</v-list-item-title>
                        <v-list-item-subtitle>
                            <v-chip>
                                <span>
                                    {{ tenant }}
                                </span>
                                <v-icon right v-clipboard="tenant" v-clipboard:success="() => { copySnack = true; }">mdi-content-copy</v-icon>
                            </v-chip>
                        </v-list-item-subtitle>
                    </v-list-item-content>
                </v-list-item>

                <v-card-actions>
                    <v-btn small text @click="logout()">
                        Logout
                    </v-btn>
                </v-card-actions>
            </v-card>
        </v-menu>
    </v-app-bar>
    <v-content>
        <v-container class="pa-8" fluid>
            <router-view></router-view>
        </v-container>
        <v-snackbar v-model="copySnack" :timeout=3000>Tenant ID copied to clipboard</v-snackbar>
    </v-content>
</v-app>
</template>

<script>
export default {
  name: "App",

  methods: {
    logout() {
      this.$store.dispatch("auth/logout").then(() => {
        this.$router.push("/login");
      });
    }
  },

  computed: {
    tenant() {
      return this.$store.getters["auth/tenant"];
    },

    isLoggedIn() {
      return this.$store.getters["auth/isLoggedIn"];
    }
  },

  data() {
    return {
      drawer: true,
      clipped: false,
      copySnack: false,
      items: [
        {
          icon: "dashboard",
          title: "Dashboard",
          path: "/"
        },
        {
          icon: "devices",
          title: "Devices",
          path: "/devices"
        },
        {
          icon: "history",
          title: "Sessions",
          path: "/sessions"
        }
      ]
    };
  }
};
</script>

<style>
.v-list-active {
  border-left: 4px solid var(--v-primary-base);
}

.text-shadow {
  text-shadow: #000 0 0 6px;
  color: transparent;
}
</style>
