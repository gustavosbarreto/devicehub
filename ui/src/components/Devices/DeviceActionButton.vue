<template>
  <div>
    <v-list-item v-if="notificationStatus">
      <v-btn
        v-bind="$attrs"
        size="x-small"
        color="primary"
        data-test="notification-btn"
        @click="dialog = !dialog"
      >
        <v-icon> {{ icon }} </v-icon>
        Accept
      </v-btn>
    </v-list-item>
    <v-list-item
      @click="dialog = !dialog"
      v-else>
      <v-tooltip location="bottom" class="text-center" :disabled="hasAuthorization">
        <template v-slot:activator="{ props }">
          <span v-bind="props">
            <v-list-item-title data-test="action-item" v-on="props">
              <v-icon> {{ icon }}</v-icon>
              {{ capitalizeText(action) }}
            </v-list-item-title>
          </span>
        </template>
        <span> You don't have this kind of authorization. </span>
      </v-tooltip>
    </v-list-item>
    <v-dialog max-width="450px" v-model="dialog" @click:outside="close" v-bind="$attrs">
      <v-card class="bg-v-theme-surface">
        <v-card-title class="text-h5 pa-5 bg-primary">
          Device {{ action }}
        </v-card-title>
        <v-divider />
        <v-container>

          <v-alert
            v-if="billingActive"
            type="warning"
            text="Accepted devices in ShellHub become active in your account and are billed for the entire billing period." />
          <v-card-text class="mt-4 mb-0 pb-1">
            <p class="mb-2">
              Do you want to {{ action }} this device?
            </p>
          </v-card-text>

          <v-card-actions>
            <v-spacer />

            <v-btn variant="text" @click="close()"> Close </v-btn>

            <v-btn variant="text" @click="doAction()">
              {{ action }}
            </v-btn>
          </v-card-actions>
        </v-container>
      </v-card>
    </v-dialog>
  </div>
</template>

<script lang="ts">
import { defineComponent, ref, computed } from "vue";
import axios, { AxiosError } from "axios";
import { useStore } from "../../store";
import { authorizer, actions } from "../../authorizer";
import hasPermission from "../../utils/permission";
import { INotificationsError } from "../../interfaces/INotifications";
import { capitalizeText } from "../../utils/string";
import handleError from "../../utils/handleError";

export default defineComponent({
  props: {
    name: {
      type: String,
      required: false,
      default: "Device",
    },
    uid: {
      type: String,
      required: true,
    },

    notificationStatus: {
      type: Boolean,
      required: false,
      default: false,
    },

    action: {
      type: String,
      default: "accept",
      validator: (value: string) => ["accept", "reject", "remove"].includes(value),
    },

    show: {
      type: Boolean,
      default: false,
      required: false,
    },
  },
  setup(props, ctx) {
    const store = useStore();

    const billingActive = computed(() => store.getters["billing/active"]);

    const hasAuthorization = computed(() => {
      const role = store.getters["auth/role"];
      if (role !== "") {
        return hasPermission(
          authorizer.role[role],
          actions.device[props.action],
        );
      }

      return false;
    });

    const dialog = ref(false);

    const close = () => {
      dialog.value = false;
      ctx.emit("update:show", false);
    };

    const refreshStats = async () => {
      try {
        await store.dispatch("stats/get");
      } catch (error: unknown) {
        store.dispatch("snackbar/showSnackbarErrorDefault");
        handleError(error);
      }
    };

    const refreshDevices = () => {
      try {
        ctx.emit("update");
        if (
          window.location.pathname === "/devices/pending"
          || window.location.pathname === "/devices"
        ) {
          store.dispatch("devices/refresh");
          store.dispatch("notifications/fetch");
        }

        close();
      } catch (error: unknown) {
        store.dispatch(
          "snackbar/showSnackbarErrorLoading",
          INotificationsError.deviceList,
        );
        handleError(error);
      }
    };

    const removeDevice = async () => {
      try {
        await store.dispatch("devices/remove", props.uid);
        refreshDevices();
      } catch (error: unknown) {
        close();

        store.dispatch(
          "snackbar/showSnackbarErrorAction",
          INotificationsError.deviceDelete,
        );
        handleError(error);
      }
    };

    const rejectDevice = async () => {
      try {
        await store.dispatch("devices/reject", props.uid);
        refreshStats();
        refreshDevices();
      } catch (error: unknown) {
        close();

        store.dispatch(
          "snackbar/showSnackbarErrorAction",
          INotificationsError.deviceRejecting,
        );
        handleError(error);
      }
    };

    const acceptDevice = async () => {
      try {
        await store.dispatch("devices/accept", props.uid);
        refreshStats();
        refreshDevices();
      } catch (error: unknown) {
        if (axios.isAxiosError(error)) {
          const axiosError = error as AxiosError;
          switch (axiosError.response?.status) {
            case 402:
              store.dispatch("users/setStatusUpdateAccountDialogByDeviceAction", true);
              break;
            case 409:
              store.dispatch("devices/setDeviceToBeRenamed", props.name);
              store.dispatch("users/setDeviceDuplicationOnAcceptance", true);
              break;
            default:
              return;
          }
        }
        close();

        store.dispatch(
          "snackbar/showSnackbarErrorAction",
          INotificationsError.deviceAccepting,
        );
        handleError(error);
      }
    };

    const doAction = () => {
      if (hasAuthorization.value) {
        switch (props.action) {
          case "accept":
            acceptDevice();
            break;
          case "reject":
            rejectDevice();
            break;
          case "remove":
            removeDevice();
            break;
          default:
        }
      } else {
        store.dispatch("snackbar/showSnackbarErrorAssociation");
      }
    };

    const findIcon = () => {
      switch (props.action) {
        case "accept":
          return "mdi-check";
        case "reject":
          return "mdi-close";
        case "remove":
          return "mdi-delete";
        default:
          return "";
      }
    };

    const icon = ref(findIcon());

    return {
      icon,
      doAction,
      close,
      hasAuthorization,
      capitalizeText,
      dialog,
      billingActive,
    };
  },
});
</script>

<style scoped>
p {
  font-size: 1rem;
}
</style>
