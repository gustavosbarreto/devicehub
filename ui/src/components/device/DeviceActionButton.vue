<template>
  <fragment>
    <fragment
      v-if="notificationStatus"
    >
      <v-btn
        x-small
        color="primary"
        data-test="notification-btn"
      >
        Accept
      </v-btn>
    </fragment>

    <fragment
      v-else
    >
      <v-tooltip
        bottom
        :disabled="hasAuthorization"
      >
        <template #activator="{ on }">
          <span v-on="on">
            <v-list-item-title
              data-test="action-item"
              v-on="on"
            >
              {{ action | capitalizeFirstLetter }}
            </v-list-item-title>
          </span>

          <span v-on="on">
            <v-icon
              :disabled="!hasAuthorization"
              left
              data-test="action-icon"
              v-on="on"
            >
              {{ icon }}
            </v-icon>
          </span>
        </template>

        <span v-if="!hasAuthorization">
          You don't have this kind of authorization.
        </span>
      </v-tooltip>
    </fragment>

    <v-dialog
      v-model="showDialog"
      max-width="400"
      @click:outside="close"
    >
      <v-card data-test="deviceActionButton-card">
        <v-card-title class="headline grey lighten-2 text-center">
          Are you sure?
        </v-card-title>

        <v-card-text class="mt-4 mb-3 pb-1">
          <div>
            You are about to {{ action }} this device.
          </div>
        </v-card-text>

        <v-card-actions>
          <v-spacer />

          <v-btn
            text
            data-test="cancel-btn"
            @click="close()"
          >
            Cancel
          </v-btn>

          <v-btn
            text
            data-test="dialog-btn"
            @click="doAction();"
          >
            {{ action }}
          </v-btn>
        </v-card-actions>
      </v-card>
    </v-dialog>
  </fragment>
</template>

<script>

import capitalizeFirstLetter from '@/components/filter/string';
import hasPermission from '@/components/filter/permission';

export default {
  name: 'DeviceActionButtonComponent',

  filters: { capitalizeFirstLetter, hasPermission },

  props: {
    uid: {
      type: String,
      required: true,
    },

    notificationStatus: {
      type: Boolean,
      required: false,
      default: false,
    },

    action: {
      type: String,
      default: 'accept',
      validator: (value) => ['accept', 'reject', 'remove'].includes(value),
    },

    show: {
      type: Boolean,
      required: true,
    },
  },

  data() {
    return {
      icon: this.findIcon(),
    };
  },

  computed: {
    showDialog: {
      get() {
        return this.show && this.hasAuthorization;
      },
      set(value) {
        this.$emit('update:show', value);
      },
    },

    hasAuthorization() {
      const role = this.$store.getters['auth/role'];
      if (role !== '') {
        return hasPermission(
          this.$authorizer.role[role],
          this.$actions.device[this.action],
        );
      }

      return false;
    },
  },

  methods: {
    doAction() {
      switch (this.action) {
      case 'accept':
        this.acceptDevice();
        break;
      case 'reject':
        this.rejectDevice();
        break;
      case 'remove':
        this.removeDevice();
        break;
      default:
      }
    },

    async acceptDevice() {
      try {
        await this.$store.dispatch('devices/accept', this.uid);
        this.refreshStats();
        this.refreshDevices();
      } catch (error) {
        if (error.response.status === 402) {
          this.$store.dispatch('users/setStatusUpdateAccountDialogByDeviceAction', true);
        }
        this.close();

        this.$store.dispatch('snackbar/showSnackbarErrorAction', this.$errors.snackbar.deviceAccepting);
      }
    },

    async rejectDevice() {
      try {
        await this.$store.dispatch('devices/reject', this.uid);
        this.refreshStats();
        this.refreshDevices();
      } catch {
        this.close();

        this.$store.dispatch('snackbar/showSnackbarErrorAction', this.$errors.snackbar.deviceRejecting);
      }
    },

    async removeDevice() {
      try {
        await this.$store.dispatch('devices/remove', this.uid);
        this.refreshDevices();
      } catch {
        this.close();

        this.$store.dispatch('snackbar/showSnackbarErrorAction', this.$errors.snackbar.deviceDelete);
      }
    },

    refreshDevices() {
      try {
        this.$emit('update');
        if (window.location.pathname === '/devices/pending' || window.location.pathname === '/devices') {
          this.$store.dispatch('devices/refresh');
          this.$store.dispatch('notifications/fetch');
        }

        this.close();
      } catch {
        this.$store.dispatch('snackbar/showSnackbarErrorLoading', this.$errors.snackbar.deviceList);
      }
    },

    async refreshStats() {
      try {
        await this.$store.dispatch('stats/get');
      } catch {
        this.$store.dispatch('snackbar/showSnackbarErrorDefault');
      }
    },

    findIcon() {
      switch (this.action) {
      case 'accept':
        return 'mdi-check';
      case 'reject':
        return 'close';
      case 'remove':
        return 'delete';
      default:
        return '';
      }
    },

    close() {
      this.$emit('update:show', false);
    },
  },
};

</script>
