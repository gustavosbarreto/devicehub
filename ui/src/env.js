export default {
  /* eslint-disable no-param-reassign */
  install(Vue) {
    const env = {
      isHosted: window.env.HOSTED === 'true',
    };

    Vue.env = env;
    Vue.prototype.$env = env;
  },
  /* eslint-enable no-param-reassign */
};
