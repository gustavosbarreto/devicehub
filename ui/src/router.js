import Vue from "vue";
import Router from "vue-router";
import Dashboard from "./views/Dashboard.vue";
import store from './store'

Vue.use(Router);

const router = new Router({
    mode: "history",
    base: process.env.BASE_URL,
    routes: [
        {
            path: "/",
            name: "dashboard",
            component: Dashboard
        },
        {
            path: "/devices",
            name: "devices",

            component: () =>
                import(/* webpackChunkName: "devices" */ "./views/Devices.vue")
        },
        {
            path: "/devices/:id",
            name: "featureDevice",
            component: () =>
                import("./views/FeatureDevice.vue") 
        },
        {
            path: "/sessions",
            name: "sessions",
            component: () =>
                import("./views/Sessions.vue")
        },
        {
            path: "/login",
            name: "login",
            component: () =>
                import("./views/Login.vue")
        }
    ]
});

router.beforeEach((to, from, next) => {
    if (to.path !== '/login') {
        if (store.getters['auth/isLoggedIn']) {
          return next()
        }

        return next(`/login?redirect=${to.path}`)
      } else {
        return next()
      }
})

export default router;