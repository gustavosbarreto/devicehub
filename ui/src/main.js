import Vue from 'vue';

import Snackbar from '@/components/snackbar/Snackbar';

import App from './App';
import router from './router';
import store from './store';
import env from './env';
import './vee-validate';
import vuetify from './plugins/vuetify';

Vue.config.productionTip = false;

Vue.component('Snackbar', Snackbar);

Vue.use(require('vue-moment'));

Vue.use(env);

new Vue({
  vuetify,
  router,
  store,
  render: (h) => h(App),
}).$mount('#app');
