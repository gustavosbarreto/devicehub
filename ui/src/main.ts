import { createApp } from "vue";
import * as Sentry from "@sentry/vue";
import { BrowserTracing } from "@sentry/tracing";
import { envVariables } from "./envVariables";
import vuetify from "./plugins/vuetify";
import { key, store } from "./store";
import { router } from "./router";
import App from "./App.vue";

import { loadFonts } from "./plugins/webfontloader";

import SnackbarComponent from "./components/Snackbar/Snackbar.vue";
import { SnackbarPlugin } from "./plugins/snackbar";

const app = createApp(App);

Sentry.init({
  app,
  dsn: envVariables.sentryDsn || "",
  integrations: [
    new BrowserTracing({
      routingInstrumentation: Sentry.vueRouterInstrumentation(router),
    }),
  ],
  tracesSampleRate: 1.0,
  hooks: ["activate", "create", "destroy", "mount", "update"],
  timeout: 500,
  release: envVariables.version || "latest",
});
Sentry.setTag("project", "shellhub-ui");

loadFonts();

app.use(vuetify);
app.use(router);
app.use(store, key);
app.use(SnackbarPlugin);
app.component("SnackbarComponent", SnackbarComponent);
app.mount("#app");
