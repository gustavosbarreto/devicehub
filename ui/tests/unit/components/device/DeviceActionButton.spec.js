import Vuex from 'vuex';
import { mount, createLocalVue } from '@vue/test-utils';
// import flushPromises from 'flush-promises';
import Vuetify from 'vuetify';
import DeviceActionButton from '@/components/device/DeviceActionButton';
import { actions, authorizer } from '../../../../src/authorizer';

describe('DeviceActionButton', () => {
  const localVue = createLocalVue();
  const vuetify = new Vuetify();
  localVue.use(Vuex);

  document.body.setAttribute('data-app', true);

  let wrapper;

  const role = ['owner', 'observer'];

  const hasAuthorization = {
    owner: true,
    observer: false,
  };

  const tests = [
    {
      description: 'Create button in the notification',
      variables: {
        isActive: true,
      },
      props: {
        uid: 'xxxxxxxx',
        notificationStatus: true,
        action: 'accept',
        show: false,
      },
      data: {
        icon: 'mdi-check',
      },
      template: {
        'notification-btn': true,
        'action-item': false,
        'action-icon': false,
        'deviceActionButton-card': false,
        'cancel-btn': false,
        'dialog-btn': false,
      },
    },
    {
      description: 'Create button in the list',
      variables: {
        isActive: true,
      },
      props: {
        uid: 'xxxxxxxx',
        notificationStatus: false,
        action: 'accept',
        show: false,
      },
      data: {
        icon: 'mdi-check',
      },
      template: {
        'notification-btn': false,
        'action-item': true,
        'action-icon': true,
        'deviceActionButton-card': false,
        'cancel-btn': false,
        'dialog-btn': false,
      },
    },
    {
      description: 'Reject button in the list',
      variables: {
        isActive: true,
      },
      props: {
        uid: 'xxxxxxxx',
        notificationStatus: false,
        action: 'reject',
        show: false,
      },
      data: {
        icon: 'close',
      },
      template: {
        'notification-btn': false,
        'action-item': true,
        'action-icon': true,
        'deviceActionButton-card': false,
        'cancel-btn': false,
        'dialog-btn': false,
      },
    },
    {
      description: 'Remove button in the list',
      variables: {
        isActive: true,
      },
      props: {
        uid: 'xxxxxxxx',
        notificationStatus: false,
        action: 'remove',
        show: false,
      },
      data: {
        icon: 'delete',
      },
      template: {
        'notification-btn': false,
        'action-item': true,
        'action-icon': true,
        'deviceActionButton-card': false,
        'cancel-btn': false,
        'dialog-btn': false,
      },
    },
    {
      description: 'Dialog',
      variables: {
        isActive: true,
      },
      props: {
        uid: 'xxxxxxxx',
        notificationStatus: false,
        action: 'accept',
        show: true,
      },
      data: {
        icon: 'mdi-check',
      },
      template: {
        'notification-btn': false,
        'deviceActionButton-card': true,
        'cancel-btn': true,
        'dialog-btn': true,
      },
    },
  ];

  const storeVuex = (isActive, currentrole) => new Vuex.Store({
    namespaced: true,
    state: {
      isActive,
      currentrole,
    },
    getters: {
      isActive: (state) => state.isActive,
      'auth/role': (state) => state.currentrole,
    },
    actions: {
      'devices/refresh': () => {},
      'devices/accept': () => {},
      'users/setStatusUpdateAccountDialog': () => {},
      'devices/reject': () => {},
      'devices/remove': () => {},
      'notifications/fetch': () => {},
      'stats/get': () => {},
      'snackbar/showSnackbarErrorDefault': () => {},
      'snackbar/showSnackbarErrorAction': () => {},
      'snackbar/showSnackbarErrorLoading': () => {},
    },
  });

  tests.forEach((test) => {
    role.forEach((currentrole) => {
      describe(`${test.description} ${currentrole}`, () => {
        beforeEach(() => {
          wrapper = mount(DeviceActionButton, {
            store: storeVuex(test.variables.isActive, currentrole),
            localVue,
            stubs: ['fragment'],
            propsData: {
              uid: test.props.uid,
              notificationStatus: test.props.notificationStatus,
              action: test.props.action,
              show: test.props.show,
            },
            vuetify,
            mocks: {
              $authorizer: authorizer,
              $actions: actions,
            },
          });
        });

        ///////
        // Component Rendering
        //////

        it('Is a Vue instance', () => {
          expect(wrapper).toBeTruthy();
        });
        it('Renders the component', () => {
          expect(wrapper.html()).toMatchSnapshot();
        });

        ///////
        // Data checking
        //////

        it('Receive data in props', () => {
          Object.keys(test.props).forEach((item) => {
            expect(wrapper.vm[item]).toEqual(test.props[item]);
          });
        });
        it('Compare data with default value', () => {
          Object.keys(test.data).forEach((item) => {
            expect(wrapper.vm[item]).toEqual(test.data[item]);
          });
        });
        it('Process data in the computed', () => {
          if (!(test.props.action === 'remove' && currentrole === 'operator')) {
            expect(wrapper.vm.hasAuthorization).toEqual(hasAuthorization[currentrole]);
          } else {
            expect(wrapper.vm.hasAuthorization).toEqual(false);
          }
        });

        //////
        // HTML validation
        //////

        it('Renders the template with data', () => {
          Object.keys(test.template).forEach((item) => {
            if (!hasAuthorization[currentrole] && currentrole === 'observer' && test.props.show) {
              expect(wrapper.find(`[data-test="${item}"]`).exists()).toBe(false);
            } else {
              expect(wrapper.find(`[data-test="${item}"]`).exists()).toBe(test.template[item]);
            }
          });
        });
      });
    });
  });
});
