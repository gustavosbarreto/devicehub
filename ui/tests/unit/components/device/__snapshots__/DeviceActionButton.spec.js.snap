// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceActionButton Create button in the list observer Renders the component 1`] = `
<fragment-stub>
  <fragment-stub><span><div data-test="action-item" class="v-list-item__title">
            accept
          </div></span><span class="v-tooltip v-tooltip--bottom"><!----></span><span><i aria-hidden="true" data-test="action-icon" class="v-icon notranslate v-icon--disabled v-icon--left mdi mdi-check theme--light"></i></span> </fragment-stub>
  <div role="dialog" class="v-dialog__container">
    <!---->
  </div>
</fragment-stub>
`;

exports[`DeviceActionButton Create button in the list owner Renders the component 1`] = `
<fragment-stub>
  <fragment-stub><span><div data-test="action-item" class="v-list-item__title">
            accept
          </div></span><span class="v-tooltip v-tooltip--bottom"><!----></span><span><i aria-hidden="true" data-test="action-icon" class="v-icon notranslate v-icon--left mdi mdi-check theme--light"></i></span> </fragment-stub>
  <div role="dialog" class="v-dialog__container">
    <!---->
  </div>
</fragment-stub>
`;

exports[`DeviceActionButton Create button in the notification observer Renders the component 1`] = `
<fragment-stub>
  <fragment-stub><button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--x-small primary" data-test="notification-btn"><span class="v-btn__content">
      Accept
    </span></button></fragment-stub>
  <div role="dialog" class="v-dialog__container">
    <!---->
  </div>
</fragment-stub>
`;

exports[`DeviceActionButton Create button in the notification owner Renders the component 1`] = `
<fragment-stub>
  <fragment-stub><button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--x-small primary" data-test="notification-btn"><span class="v-btn__content">
      Accept
    </span></button></fragment-stub>
  <div role="dialog" class="v-dialog__container">
    <!---->
  </div>
</fragment-stub>
`;

exports[`DeviceActionButton Dialog observer Renders the component 1`] = `
<fragment-stub>
  <fragment-stub><span><div data-test="action-item" class="v-list-item__title">
            accept
          </div></span><span class="v-tooltip v-tooltip--bottom"><!----></span><span><i aria-hidden="true" data-test="action-icon" class="v-icon notranslate v-icon--disabled v-icon--left mdi mdi-check theme--light"></i></span> </fragment-stub>
  <div role="dialog" class="v-dialog__container">
    <!---->
  </div>
</fragment-stub>
`;

exports[`DeviceActionButton Dialog owner Renders the component 1`] = `
<fragment-stub>
  <fragment-stub><span><div data-test="action-item" class="v-list-item__title">
            accept
          </div></span><span class="v-tooltip v-tooltip--bottom"><!----></span><span><i aria-hidden="true" data-test="action-icon" class="v-icon notranslate v-icon--left mdi mdi-check theme--light"></i></span> </fragment-stub>
  <div role="dialog" class="v-dialog__container"></div>
</fragment-stub>
`;

exports[`DeviceActionButton Reject button in the list observer Renders the component 1`] = `
<fragment-stub>
  <fragment-stub><span><div data-test="action-item" class="v-list-item__title">
            reject
          </div></span><span class="v-tooltip v-tooltip--bottom"><!----></span><span><i aria-hidden="true" data-test="action-icon" class="v-icon notranslate v-icon--disabled v-icon--left material-icons theme--light">close</i></span> </fragment-stub>
  <div role="dialog" class="v-dialog__container">
    <!---->
  </div>
</fragment-stub>
`;

exports[`DeviceActionButton Reject button in the list owner Renders the component 1`] = `
<fragment-stub>
  <fragment-stub><span><div data-test="action-item" class="v-list-item__title">
            reject
          </div></span><span class="v-tooltip v-tooltip--bottom"><!----></span><span><i aria-hidden="true" data-test="action-icon" class="v-icon notranslate v-icon--left material-icons theme--light">close</i></span> </fragment-stub>
  <div role="dialog" class="v-dialog__container">
    <!---->
  </div>
</fragment-stub>
`;

exports[`DeviceActionButton Remove button in the list observer Renders the component 1`] = `
<fragment-stub>
  <fragment-stub><span><div data-test="action-item" class="v-list-item__title">
            remove
          </div></span><span class="v-tooltip v-tooltip--bottom"><!----></span><span><i aria-hidden="true" data-test="action-icon" class="v-icon notranslate v-icon--disabled v-icon--left material-icons theme--light">delete</i></span> </fragment-stub>
  <div role="dialog" class="v-dialog__container">
    <!---->
  </div>
</fragment-stub>
`;

exports[`DeviceActionButton Remove button in the list owner Renders the component 1`] = `
<fragment-stub>
  <fragment-stub><span><div data-test="action-item" class="v-list-item__title">
            remove
          </div></span><span class="v-tooltip v-tooltip--bottom"><!----></span><span><i aria-hidden="true" data-test="action-icon" class="v-icon notranslate v-icon--left material-icons theme--light">delete</i></span> </fragment-stub>
  <div role="dialog" class="v-dialog__container">
    <!---->
  </div>
</fragment-stub>
`;
